shared/teams/emojis/add-emoji.tsx

```diff
@@ -0,0 +1,594 @@
+  emojisToAdd: Array<EmojiToAdd>,
+  setErrors: (errors: Map<string, string>) => void,
+  removeFilePath: (toRemove: Set<string> | string) => void
+) => {
+  const dispatch = Container.useDispatch()
+  const addEmojisRpc = useRPC(RPCChatGen.localAddEmojisRpcPromise)
+  const [waitingAddEmojis, setWaitingAddEmojis] = React.useState(false)
+  const [bannerError, setBannerError] = React.useState('')
+  const doAddEmojis =
+    conversationIDKey !== ChatConstants.noConversationIDKey
+      ? () => {
+          setWaitingAddEmojis(true)
+          addEmojisRpc(
+            [
+              {
+                aliases: emojisToAdd.map(e => e.alias),
+                convID: ChatTypes.keyToConversationID(conversationIDKey),
+                filenames: emojisToAdd.map(e => e.path),
+              },
+            ],
+            res => {
+              const failedFilenamesKeys = Object.keys(res.failedFilenames || {})
+
+              if (!failedFilenamesKeys.length) {
+                dispatch(RouteTreeGen.createClearModals())
+              }
+
+              res.successFilenames && removeFilePath(new Set(res.successFilenames))
+              setErrors(new Map(failedFilenamesKeys.map(key => [key, res.failedFilenames[key]])))
+              setBannerError(
+                `Failed to add ${failedFilenamesKeys.length} ${pluralize(
+                  'emojis',
+                  failedFilenamesKeys.length
+                )}.`
+              )
+              setWaitingAddEmojis(false)
+            },
+            err => {
+              throw err
+            }
+          )
+        }
+      : undefined
+  return {bannerError, doAddEmojis, waitingAddEmojis}
+}
+
+const useStuff = (conversationIDKey: ChatTypes.ConversationIDKey) => {
+  const [filePaths, setFilePaths] = React.useState<Array<string>>([])
+
+  const aliasMap = React.useRef(new Map<string, string>()).current
+  const [aliasMapChangeCounter, setAliasMapChangeCounter] = React.useState(0)
+  const aliasMapChanged = React.useCallback(() => setAliasMapChangeCounter(c => c + 1), [
+    setAliasMapChangeCounter,
+  ])
+
+  const addFiles = React.useCallback(
+    (paths: Array<string>) => {
+      paths.forEach(path => aliasMap.get(path) ?? aliasMap.set(path, filePathToDefaultAlias(path)))
+      aliasMapChanged()
+      setFilePaths([...filePaths, ...paths])
+    },
+    [filePaths, aliasMap, aliasMapChanged, setFilePaths]
+  )
+  const clearFiles = React.useCallback(() => setFilePaths([]), [setFilePaths])
+
+  const removeFilePath = React.useCallback(
+    (toRemove: Set<string> | string) =>
+      setFilePaths(filePaths =>
+        typeof toRemove === 'string'
+          ? filePaths.filter(filePath => toRemove !== filePath)
+          : filePaths.filter(filePath => !toRemove.has(filePath))
+      ),
+    [setFilePaths]
+  )
+
+  const [errors, setErrors] = React.useState(new Map<string, string>())
+
+  const emojisToAdd = React.useMemo(() => {
+    // @ts-ignore
```

Comment:
why

Reply:
I want this to update on `aliasMapChangeCounter` change but the linter doesn't like the fact that it's not used inside the function. I didn't want to add `@ts-ignore` on the dependency array because I still want that check to yell at me if I somehow mess up other dependencies.